packages/fetchai/protocols/default/message.py

```diff
@@ -250,13 +250,6 @@ def _is_consistent(self) -> bool:
                         self.target
                     ),
                 )
-            else:
-                enforce(
-                    0 < self.target < self.message_id,
-                    "Invalid 'target'. Expected an integer between 1 and {} inclusive. Found {}.".format(
-                        self.message_id - 1, self.target,
-                    ),
-                )
```

Comment:
We should probably extract the whole check (lines 245-259) into the message base class, use absolute value and then just call it here. That way we also avoid code duplication.

Reply:
would be nice to solve it in another PR with generator/base message class refactoring